compiler/rustc_lint/src/context.rs

```diff
@@ -353,10 +353,21 @@ impl LintStore {
             lint_name.to_string()
         };
         // If the lint was scoped with `tool::` check if the tool lint exists
-        if tool_name.is_some() {
+        if let Some(tool_name) = tool_name {
             match self.by_name.get(&complete_name) {
                 None => match self.lint_groups.get(&*complete_name) {
-                    None => return CheckLintNameResult::Tool(Err((None, String::new()))),
+                    // If the lint isn't registered, there are two possibilities:
+                    None => {
+                        // 1. The tool is currently running, so this lint really doesn't exist.
+                        // FIXME: should this handle tools that never register a lint, like rustfmt?
+                        return if self.by_name.keys().any(|lint| lint.starts_with(&format!("{}::", tool_name))) {
```

Comment:
Could the `format` be moved out of the loop here?

Reply:
Done, thanks.